### ISSUE TITLE
Downloading .bin file, How can download PDF file, I pass the url of PDF file

### ISSUE THREAD

**Issue by korat2511** [abdallah-odeh/flutter_file_downloader#57]
How can download PDF file, I pass the url of PDF file.

           IconButton(
                  icon: Icon(Icons.file_download),
                  onPressed: () {
                    log("url == $fileUrl");

                    FileDownloader.downloadFile(
                      
                        url: fileUrl,
                    
                        onDownloadCompleted: (String path) {
                          print('FILE DOWNLOADED TO PATH: $path');
                        },
                        onDownloadError: (String error) {
                          print('DOWNLOAD ERROR: $error');
                        });
                  },
                ),

**Comment by abdallah-odeh** [comment]
Hello @korat2511, 
Downloading any file type follows the same procedure, 
Did you try to download the file using the example project?

**Comment by korat2511** [comment]
Yes, I tried but not working

**Comment by abdallah-odeh** [comment]
Can you please provide me with the link you are trying to download

**Comment by korat2511** [comment]
https://pmcprojects.in/daily-progress-report?site=70&date=2024-09-19&category_id=16,8,14,7,15,18,2,4,1,17,5,13,3,6&user_id=1,6&photos=1&task=Pending,Active,Complete,Overdue&survey=1&material=1&manpower=1&work_update=1&decision=1

**Comment by abdallah-odeh** [comment]
The URL returns status code 500

**Comment by preetpatel21** [comment]
I am facing same issue.
this is the url
https://assets.crewlair.com/chat/7fd6819b-42e6-4795-97fa-7ef9549d6d20/image/2024-10-09T06%3A51%3A04.096966Z

**Comment by abdallah-odeh** [comment]
Hello @preetpatel21, 
I tried the URL you provided, indeed there is an issue, when the URL does not contain the file extension the file gets downloaded without an extension, so the OS cannot open it.

I will be working on this bug this weekend inshallah

**Comment by abdallah-odeh** [comment]
Hello @korat2511,
Is there any update in your case? do face the same issue? (The file gets downloaded but you cannot open/view it)
or the file does not download from the beginning

**Comment by korat2511** [comment]
Yes, i am facing same issue

**Comment by abdallah-odeh** [comment]
Hello @korat2511 & @preetpatel21 
Can you please try the new prerelease version `flutter_file_downloader: ^2.0.1-dev.3` and tell me the results you get

**Comment by preetpatel21** [comment]
@abdallah-odeh after upgrading to the ^2.0.1-dev.3 version , i can able to download that file.

**Comment by korat2511** [comment]
![Screenshot 2024-10-16 at 11 56 50 PM](https://github.com/user-attachments/assets/87fc8642-a131-491d-b85f-f4d97dcffe87)


Now  i am facing this issue

**Comment by korat2511** [comment]
And i am still getting bin file 
here is url 

[log] url == https://pmcprojects.in/daily-progress-report?site=61&date=2024-10-16&category_id=16,19,8,27,14,7,15,25,18,2,4,22,1,20,17,5,21,13,26,3,6,23,24&user_id=1,2,6&photos=1&task=Pending,Active,Complete,Overdue&survey=1&material=1&manpower=1&work_update=1&decision=1&drawing=1

**Comment by korat2511** [comment]
Any update on this issue, I am still getting .bin file

**Comment by abdallah-odeh** [comment]
Did you follow the steps mentioned in the other issue? #60 
because I tested it on the sample app and it worked fine.

**Comment by korat2511** [comment]
Yes, now it is working perfectly for Android, do you have solution for iOS ?

**Comment by abdallah-odeh** [comment]
well, I've done no changes 😅
the plugin obviously mentions that it only supports android
